examples/jsm/nodes/display/DotScreenNode.js

```diff
@@ -0,0 +1,131 @@
+import TempNode from '../core/TempNode.js';
+import { nodeObject, addNodeElement, tslFn, vec2, vec3, vec4 } from '../shadernode/ShaderNode.js';
+import { uniform } from '../core/UniformNode.js';
+import { NodeUpdateType } from '../core/constants.js';
+import { uv } from '../accessors/UVNode.js';
+import { texturePass } from './PassNode.js';
+import { sin, cos } from '../math/MathNode.js';
+import { add } from '../math/OperatorNode.js';
+import { Vector2, RenderTarget } from 'three';
+import QuadMesh from '../../objects/QuadMesh.js';
+
+const quadMesh = new QuadMesh();
+
+class DotScreenNode extends TempNode {
+
+	constructor( textureNode, center = new Vector2( 0.5, 0.5 ), angle = 1.57, scale = 1 ) {
+
+		super( 'vec4' );
+
+		this.textureNode = textureNode;
+		this.center = uniform( center );
+		this.angle = uniform( angle );
+		this.scale = uniform( scale );
+
+		this._renderTarget = new RenderTarget();
+		this._renderTarget.texture.name = 'dotScreen';
+
+		this._size = uniform( new Vector2() );
+
+		this._textureNode = texturePass( this, this._renderTarget.texture );
+
+		this.updateBeforeType = NodeUpdateType.RENDER;
+
+	}
+
+	getTextureNode() {
+
+		return this._textureNode;
+
+	}
+
+	setSize( width, height ) {
+
+		this._size.value.set( width, height );
+		this._renderTarget.setSize( width, height );
+
+	}
+
+	updateBefore( frame ) {
+
+		const { renderer } = frame;
+
+		const textureNode = this.textureNode;
+		const map = textureNode.value;
+
+		this._renderTarget.texture.type = map.type;
+
+		const currentRenderTarget = renderer.getRenderTarget();
+		const currentTexture = textureNode.value;
+
+		quadMesh.material = this._material;
+
+		this.setSize( map.image.width, map.image.height );
+
+		// render
+
+		renderer.setRenderTarget( this._renderTarget );
+
+		quadMesh.render( renderer );
+
+		// restore
+
+		renderer.setRenderTarget( currentRenderTarget );
+		textureNode.value = currentTexture;
+
+	}
+
+	setup( builder ) {
+
+		const { textureNode } = this;
+
+		const uvNode = textureNode.uvNode || uv();
+
+		const sampleTexture = ( uv ) => this.textureNode.uv( uv );
+
+		const pattern = tslFn( () => {
+
+			const s = sin( this.angle );
+			const c = cos( this.angle );
+
+			const tex = uvNode.mul( this._size ).sub( this.center );
+			const point = vec2( c.mul( tex.x ).sub( s.mul( tex.y ) ), s.mul( tex.x ).add( c.mul( tex.y ) ) ).mul( this.scale );
+
+			return sin( point.x ).mul( sin( point.y ) ).mul( 4 );
+
+		} );
+
+		const dotScreen = tslFn( () => {
+
+			const color = sampleTexture( uvNode );
+
+			const average = add( color.r, color.g, color.b ).div( 3 );
+
+			return vec4( vec3( average.mul( 10 ).sub( 5 ).add( pattern() ) ), color.a );
+
+		} );
+
+		//
+
+		const material = this._material || ( this._material = builder.createNodeMaterial() );
+		material.fragmentNode = dotScreen();
+
+		//
+
+		const properties = builder.getNodeProperties( this );
+		properties.textureNode = textureNode;
+
+		//
+
+		return this._textureNode;
```

Comment:
When https://github.com/mrdoob/three.js/issues/28754 is solved, `DotScreenNode` would return `dotScreen()` if no texture is required.

Reply:
I think I understand the issue, we could solve it with an intermediate node that transforms the Node output into texture, similar to RTTNode to perform samples. When we can do:

```js
let textureNode = this.textureNode;

if ( textureNode.isTextureNode !== true ) textureNode = textureNode.toRTT() // it will return a TextureNode extended class
```

Reply:
Yes, but we have to make sure to do this when the pass _after_ `DotScreenNode` requires a texture. When I understand the code correctly, `textureNode` is set from the previous pass, right?

Reply:
> Yes, but we have to make sure to do this when the pass after DotScreenNode requires a texture. When I understand the code correctly, textureNode is set from the previous pass, right?

That's why we need a RTTNode, this should solve this issue. I would suggest merge this PR, and I will make an update soon, this will be useful as part of the testing.

Reply:
Okay, let's go that route!